crates/bridge-registry-backends/src/dkg.rs

```diff
@@ -0,0 +1,63 @@
+use webb::substrate::subxt::{OnlineClient, PolkadotConfig};
+use webb::substrate::dkg_runtime::api as RuntimeApi;
+use webb::substrate::dkg_runtime::api::runtime_types::pallet_bridge_registry::types::BridgeMetadata;
+use webb_proposals::ResourceId;
+
+pub struct DkgBridgeRegistryBackend {
+    client: OnlineClient<PolkadotConfig>,
+}
+
+impl DkgBridgeRegistryBackend {
+    pub fn new(client: OnlineClient<PolkadotConfig>) -> Self {
+        Self { client }
+    }
+}
+
+#[async_trait::async_trait]
+impl super::BridgeRegistryBackend for DkgBridgeRegistryBackend {
+    async fn next_bridge_index(&self) -> webb_relayer_utils::Result<u32> {
+        let storage = RuntimeApi::storage().bridge_registry();
+        let next_bridge_index = storage.next_bridge_index();
+        Ok(self
+            .client
+            .storage()
+            .at(None)
+            .await?
+            .fetch(&next_bridge_index)
+            .await?
+            .unwrap_or(1))
+    }
+
+    async fn resource_to_bridge_id(
```

Comment:
Let's be more clear about naming methods, this uses an inner method called `resource_to_bridge_index` so lets use that.

```suggestion
    async fn resource_to_bridge_index(
```